### ISSUE TITLE
Review schemas for possible consolidation and clean-up

### ISSUE THREAD

**Issue by MVrachev** [theupdateframework/tuf#1043]
**Description of issue or feature request**:
`FILEDICT_SCHEMA` and `TARGETINFO_SCHEMA` are doing identical jobs but with the different representations of the information. Just look at their description:
![image](https://user-images.githubusercontent.com/16246778/83752894-f8912b00-a671-11ea-859f-4d7683e0e407.png)

![image](https://user-images.githubusercontent.com/16246778/83752868-eadba580-a671-11ea-953a-bf0681c09803.png)

This issue is based on a discussion with Joshua Lock: https://github.com/theupdateframework/tuf/pull/1031#discussion_r433781932

**Current behavior**:
`FILEDICT_SCHEMA` is used in some functions when it is suitable and `TARGETINFO_SCHEMA` in other places where it would be easier to use it.

**Expected behavior**:
I think we should stick to one of those even if it means adding some additional argument to functions and doing little refactoring.

**Comment by joshuagl** [comment]
We should review the schemas, consolidate and clean up.

FILEDICT_SCHEMA and TARGETINFO_SCHEMA are clearly serving different purposes, apologies for the bad lead.

Let's review schemas after #1031 has resolved, as that PR is renaming and cleaning up some of the schemas we are discussing here and makes it hard to reason about where overlap exists.

**Comment by jku** [comment]
Closing this issue as it was filed against (what is now known as) the legacy codebase: issue seems to not be relevant anymore. Please re-open or file a new issue if you feel that the issue is revelant to current python-tuf.

The current parsing/validation is likely not perfect but it is far easier to reason about.

### More details

Current source code (and upcoming 1.0 release) only contains the modern components
 * a low-level Metadata API (`tuf.api`) and 
 * `tuf.ngclient` that implements the client workflow, 

Legacy components (e.g. tuf.client, tuf.repository_tool, tuf.repository_lib as well as the repo and client scripts) are no longer included. See [announcement](https://github.com/theupdateframework/python-tuf/blob/develop/docs/1.0.0-ANNOUNCEMENT.md) and [API reference](https://theupdateframework.readthedocs.io/en/latest/api/api-reference.html) for more details.
